src/libraries/System.Collections/src/System/Collections/Generic/OrderedDictionary.cs

```diff
@@ -0,0 +1,1864 @@
+        /// has the default initial capacity, and uses the specified <see cref="IEqualityComparer{TKey}"/>.
+        /// </summary>
+        /// <param name="comparer">
+        /// The <see cref="IEqualityComparer{TKey}"/> implementation to use when comparing keys,
+        /// or null to use the default <see cref="EqualityComparer{TKey}"/> for the type of the key.
+        /// </param>
+        public OrderedDictionary(IEqualityComparer<TKey>? comparer) : this(0, comparer)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrderedDictionary{TKey, TValue}"/> class that is empty,
+        /// has the specified initial capacity, and uses the specified <see cref="IEqualityComparer{TKey}"/>.
+        /// </summary>
+        /// <param name="capacity">The initial number of elements that the <see cref="OrderedDictionary{TKey, TValue}"/> can contain.</param>
+        /// <param name="comparer">
+        /// The <see cref="IEqualityComparer{TKey}"/> implementation to use when comparing keys,
+        /// or null to use the default <see cref="EqualityComparer{TKey}"/> for the type of the key.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">capacity is less than 0.</exception>
+        public OrderedDictionary(int capacity, IEqualityComparer<TKey>? comparer)
+        {
+            ArgumentOutOfRangeException.ThrowIfNegative(capacity);
+
+            if (capacity > 0)
+            {
+                EnsureBucketsAndEntriesInitialized(capacity);
+            }
+
+            // Initialize the comparer:
+            // - Strings: Special-case EqualityComparer<string>.Default, StringComparer.Ordinal, and
+            //   StringComparer.OrdinalIgnoreCase. We start with a non-randomized comparer for improved throughput,
+            //   falling back to a randomized comparer if the hash buckets become sufficiently unbalanced to cause
+            //   more collisions than a preset threshold.
+            // - Other reference types: we always want to store a comparer instance, either the one provided,
+            //   or if one wasn't provided, the default (accessing EqualityComparer<TKey>.Default
+            //   with shared generics on every dictionary access can add measurable overhead).
+            // - Value types: if no comparer is provided, or if the default is provided, we'd prefer to use
+            //   EqualityComparer<TKey>.Default.Equals on every use, enabling the JIT to
+            //   devirtualize and possibly inline the operation.
+            if (!typeof(TKey).IsValueType)
+            {
+                _comparer = comparer ?? EqualityComparer<TKey>.Default;
+
+                if (typeof(TKey) == typeof(string) &&
+                    NonRandomizedStringEqualityComparer.GetStringComparer(_comparer!) is IEqualityComparer<string> stringComparer)
```

Comment:
Is the suppression necessary here?
```suggestion
                    NonRandomizedStringEqualityComparer.GetStringComparer(_comparer) is IEqualityComparer<string> stringComparer)
```

Reply:
It shouldn't be. I'll double check.